src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,248 @@
+                    topics = ORDER_RECEIVED_TOPIC,
+                    autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
+    public void processOrderReceived(org.springframework.messaging.Message<OrderReceived> message)
             throws SerializationException, ExecutionException, InterruptedException {
+        OrderReceived orderReceived = message.getPayload();
+        String orderReceivedUri = orderReceived.getOrderUri();
         try {
-            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC);
-        } catch (Exception ex){
-            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY, ex);
+            logMessageReceived(message, ORDER_RECEIVED_TOPIC);
+
+            logMessageProcessed(message, ORDER_RECEIVED_TOPIC);
+        } catch (ServiceException ex){
+            logMessageProcessingFailureRecoverable(message, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY, ex);
+            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY);
+        } catch (Exception x) {
+            logMessageProcessingFailureNonRecoverable(message, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY, x);
         }
     }
 
-    @KafkaListener(topics = ORDER_RECEIVED_TOPIC_RETRY, groupId = ORDER_RECEIVED_GROUP_RETRY,
-            autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
-    public void processOrderReceivedRetry(String orderReceivedUri)
-            throws SerializationException, ExecutionException, InterruptedException {
+    /**
+     * `order-received-retry` topic listener/consumer. Calls a `retryable` method to process received message.
+     * If the `retryable` processor is unsuccessful with a `retryable` error, after maximum numbers of attempts allowed,
+     * the message is published to `-error` topic for failover processing.
+     * @param message
+     * @throws SerializationException
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    @KafkaListener(id = ORDER_RECEIVED_GROUP_RETRY, groupId = ORDER_RECEIVED_GROUP_RETRY,
+                    topics = ORDER_RECEIVED_TOPIC_RETRY,
+                    autoStartup = "#{!${uk.gov.companieshouse.item-handler.error-consumer}}")
+    public void processOrderReceivedRetry(org.springframework.messaging.Message<OrderReceived> message)
+            throws InterruptedException, ExecutionException, SerializationException {
+        OrderReceived orderReceived = message.getPayload();
+        String orderReceivedUri = orderReceived.getOrderUri();
         try {
-            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC_RETRY);
-        } catch (Exception ex){
-            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR, ex);
+            logMessageReceived(message, ORDER_RECEIVED_TOPIC_RETRY);
+
+            logMessageProcessed(message, ORDER_RECEIVED_TOPIC_RETRY);
```

Comment:
Message appears to be logged twice.

Reply:
They are two separate messages. The processing of message (sending email) happens between these two messages.